### PR TITLE
Time interval redux

### PR DESCRIPTION
**Pull request by mwhite919**
everything set up except for changing the dates to hours when it switches to days.

### CODE REVIEW

File: app/components/ChartsMain.tsx [hunk 1]
```diff
@@ -117,16 +124,19 @@ export const ChartsMain = () => {
         </div>
         <div>
           These coins:{" "}
-          {combinedChartCoins.map((c) => {
-            return <div key={c.coinName}>{c.coinName}</div>;
+          {combinedChartCoins?.map((c) => {
+            return <div key={c?.coinName}>{c?.coinName}</div>;
```
Comment (yoloOnTheBattlefield) [inline app/components/ChartsMain.tsx]: do you really need the ? inside the map?

Comment (yoloOnTheBattlefield) [reply app/components/ChartsMain.tsx]: https://www.loom.com/share/7f47f5df7cd5467c9be0a4023698c7da?from_recorder=1&focus_title=1

File: app/components/CoinSwiper.tsx [hunk 2]
```diff
@@ -37,7 +35,7 @@ export const CoinSwiper = ({ handleClick, combinedChartCoins }) => {
         <div>
           <Slider {...settings}>
             {top10Coins?.map((coin) => (
-              <div key={coin.id}>
+              <div key={coin?.id}>
```
Comment (yoloOnTheBattlefield) [inline app/components/CoinSwiper.tsx]: https://www.loom.com/share/73946853086648818aeed2719ac079cb?from_recorder=1&focus_title=1

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: app/components/CoinSwiper.tsx (hunk 2 above)
Comment (mwhite919) [reply app/components/CoinSwiper.tsx]: lol I probably did a replace all. question marks solve a lot of my problems but I can't imagine any reason I would see putting it in with the key, as the solution.

**Pull request merged**